### PR TITLE
Add unquote support

### PR DESCRIPTION
**Pull request by ngeiswei**
Add/complete unquote support according to http://wiki.opencog.org/wikihome/index.php/UnquoteLink

The code is spread across multiple places. I think (but I'm not 100% sure) one can simplify all that by setting clauses as PatternTerms. But I'm letting that for another PR, if any.

### CODE REVIEW

File: opencog/atoms/pattern/PatternLink.cc [hunk 1]
```diff
@@ -589,7 +589,7 @@ void PatternLink::unbundle_virtual(const std::set<Handle>& vars,
 		bool is_virtual = false;
 		bool is_black = false;
 
-#ifdef BORKEN_DOESNT_WORK
+#ifdef BROKEN_DOESNT_WORK
```
Comment (linas) [inline opencog/atoms/pattern/PatternLink.cc]: well, if it merely broken, then its not borken.  (c.f. Robert Bork, who was borked.)

**Comment by linas** [comment]
wow that was a lot of changes.

**Comment by linas** [comment]
There is one important spot that I think you missed: `atoms/execution/Instantiator.cc`

**Comment by ngeiswei** [comment]
Fixed, thanks.

**Pull request merged**